### PR TITLE
Limit the maximum number of failed login attempts for a user.

### PR DESCRIPTION
**Pull request by andy9775**
After a specified number of failed login attempts, the user is locked out for a defined period of
time. Once the time has passed, the user can attempt to login again.

closes: 12

### CODE REVIEW

**Comment by andy9775** [comment]
Note, leaving out f/e changes as we need to standardize the error component

**Comment by plausibly** [comment]
Just took a quick look, I don't think `mockdate` is necessary. Jest allows you to mock the current time, I did something similar in the postdashboard test
https://jestjs.io/docs/jest-object#jestsetsystemtimenow-number--date
  ```
jest.useFakeTimers('modern')
jest.setSystemTime(new Date(...) )
```

**Review by danielduong8 — APPROVED** [review]
LGTM

File: server/controllers/v1/tests/user.test.ts [hunk 1]
```diff
@@ -143,6 +145,71 @@ describe('v1 - User Controller', () => {
       test('Password should not be included in serialized result', () => {
         expect(signInResponse.data.result.password).not.toBeDefined();
       });
+
```
Comment (plausibly) [inline server/controllers/v1/tests/user.test.ts:147]: Add test: Other users should not be affected if someone is locked out.

(User 1 is locked out, user 2 can still login normally)

**Pull request merged**